client/packages/system/src/Patient/Encounter/DetailModal.tsx

```diff
@@ -1,95 +1,144 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import {
+  AlertIcon,
   BasicSpinner,
+  Box,
   DialogButton,
+  InputWithLabelRow,
+  Stack,
+  Typography,
   useDialog,
 } from '@openmsupply-client/common';
+import { useTranslation } from '@common/intl';
 import { useEncounter } from './api/hooks';
 import { usePatientModalStore } from '../hooks';
 import { PatientModal } from '../PatientView';
 import { ProgramRowFragmentWithId, usePatient } from '../api';
-import { SaveDocumentMutation, useJsonForms } from '../JsonForms';
+import { CreateDocument, useJsonForms } from '../JsonForms';
 import { ProgramSearchInput } from '../Components';
 
-const useUpsertEncounter = (
-  patientId: string,
-  programType: string,
-  type: string
-): SaveDocumentMutation => {
-  const { mutateAsync: insertEncounter } = useEncounter.document.insert();
-  const { mutateAsync: updateEncounter } = useEncounter.document.update();
-  return async (jsonData: unknown, formSchemaId: string, parent?: string) => {
-    if (parent === undefined) {
-      const result = await insertEncounter({
-        data: jsonData,
-        schemaId: formSchemaId,
-        patientId,
-        type,
-        programType,
-      });
-      return result;
-    } else {
-      const result = await updateEncounter({
-        data: jsonData,
-        parent,
-        schemaId: formSchemaId,
-      });
-      return result;
-    }
-  };
-};
-
 export const EncounterDetailModal: FC = () => {
   const patientId = usePatient.utils.id();
-
-  const { current, document, programType, reset, setProgramType } =
-    usePatientModalStore();
-
+  const t = useTranslation('patients');
+  const { current, reset, document } = usePatientModalStore();
   const [program, setProgram] = useState<ProgramRowFragmentWithId | null>(null);
-  const handleSave = useUpsertEncounter(
+  const [error, setError] = useState(false);
```

Comment:
there's altogether too much state and hook calls for my liking, but I think all are required